src/main/java/org/shortest_path_problem/omsu/Path.java

```diff
@@ -1,10 +1,39 @@
 package org.shortest_path_problem.omsu;
 
 import org.shortest_path_problem.omsu.data_types.PathAndDistance;
+import org.shortest_path_problem.omsu.data_types.PointersAndDistances;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.shortest_path_problem.omsu.Algorithms.bellmanFord;
+import static org.shortest_path_problem.omsu.Algorithms.dijkstra;
 
 public class Path {
-    public static PathAndDistance path(int s, int t, double[][] graphWeightMatrix) {
-        // TODO
-        return null;
+    public static PathAndDistance pathForDijkstra(int s, int t, double[][] graphWeightMatrix) {
+        List<Integer> pathList = new ArrayList<>();
```

Comment:
Один из вариантов, который фиксит проблему с addFirst
```suggestion
        LinkedList<Integer> pathList = new LinkedList<>(graphWeightMatrix[0].length);
```
Лично мне не очень нравится, я бы предпочел не работать с LinkedList, но в общем и целом вполне допустим